_posts/en/2016-03-01-DOM-event-listening-made-easy.md

```diff
@@ -0,0 +1,71 @@
+---
+layout: *post
+
+title: DOM event listening made easy
+tip-number: 50
+tip-username: octopitus
+tip-username-profile: https://github.com/octopitus
+tip-tldr: An elegant and easy way to handle DOM events
+
+categories:
+    - en
+---
+Many of us are still doing these things:
+
+- `element.addEventListener('type', obj.method.bind(obj))`
+- `element.addEventListener('type', function (event) {})`
+- `element.addEventListener('type', (event) => {})`
+
+Closure keeps a pointer to its enclosing scope. As a result, atttaching a closure to a DOM element can create a circular reference and thus, a memory leak. Since `element` also keeps a reference to the closure, we have a cycle that won't be cleaned up by garbage collection.
+
+Unless you have addressed your handlers in a way you can remove it later on, otherwise you will never able to remove them. And there must be the correct ways:
```

Comment:
Suggest the following rewording:

The above examples all create new anonymous event handlers that can't be removed when no longer needed. This may cause performance problems or unexpected logic bugs, when handlers that you no longer need still get accidentally triggered through unexpected user interactions or [event bubbling](http://www.javascripter.net/faq/eventbubbling.htm).

Safer event-handling patterns include the following: